fe/fe-core/src/main/java/com/starrocks/qe/ConnectContext.java

```diff
@@ -517,17 +524,27 @@ public void setCurrentCatalog(String currentCatalog) {
     public void kill(boolean killConnection) {
         LOG.warn("kill timeout query, {}, kill connection: {}",
                 getMysqlChannel().getRemoteHostPortString(), killConnection);
+        // Now, cancel running process.
+        StmtExecutor executorRef = executor;
+        if (executorRef != null) {
+            executorRef.cancel();
+        }
+        while (closed == false) {
```

Comment:
why check the closed state?

Reply:
to make sure the cancel state is return to the mysql client first

Reply:
Mysql will disable auto reconnect if it receives a cancel message. 
Closed will wait until ReadListener's handleEvent function return from processOnce() and call cleanup, in cleanup function, it will set close to be true.
At the end of processOnce() function, in finalizeCommand(), it will return the cancelled state to mysql.